# Assignment 2
## Statictical Treatment For Retail Dataset

Pada Tugas 2 kali ini saya ingin melakukan analisis data dengan menggunakan dengan statictical treatment. dataset yang akan digunakan dalam assignment kali ini dataset yang bernama nyc-rolling-sales.csv. dataset ini berisi data setiap bangunan atau unit bangunan seperti apartment dll, yang terjual di New York City dalam 12 bulan terakhir. informasi mengenai dataset dapat dilihat dibawah ini.


Atribut Information Dataset :
1. Borough : kode lokasi untuk lokasi properti
2. Neighborhood :
3. Building Class Category : Kategori Poperti yang dibangun
4. Tax Class At Present : pajak kelas dari properti
5. Block : kombinasi dari borough, blok dan lot yang ada di new york
6. lot : label untuk mengetahui bangunan itu nyata
7. Easement : hak yang dapat digunakan untuk keperluan terbatas
8. Building Class At Present : Kelas dari properti yang dibangun
9. Address : Lokasi Dari Porperti
10. Apartment Number : nomor apartement
11. Zip Code : kode sip/pos
12. Residental Units : nomor residen dari unit yang terdaftar
13. commercial units : unit komersial dari properti yang terdaftar
14. total unit : total unit pada properti terdaftar
15. Land Square Feet : luas lokasi poperti yang terdaftar dalam square feed
16. Gross Square Feet: total dari seluruh lantai gedung diukur dari permukaan luar dari dinding luar gedung. termasuk lokasi area dan jarak antar gedung atau struktur didalam properti
17. Year Build : tahun properti dibangun
18. Building class at time sale : harga gedung saat di jual
19. sales price : harga penjualaan properti
20. sales date : tanggal properti terjual
21. $0 sales price : pemindahan hak milik tanpa pertimbangan tunai

Untuk mempermudah pengambilan dataset untuk analisis, saya menggunakan fitur yang sudah tersedia di google collab dan kaggle untuk menggunakan kaggle API sehingga kita tidak perlu mendownload dataset secara manual, tapi hanya menggunakan API Command untuk mengimport dataset kedalam Google Collab.

In [ ]:
from google.colab import files
files.upload()
! mkdir ~/.kaggle
! cp kaggle.json ~/.kaggle/
! chmod 600 ~/.kaggle/kaggle.json

Saving kaggle.json to kaggle.json


In [ ]:
! kaggle datasets download -d new-york-city/nyc-property-sales

nyc-property-sales.zip: Skipping, found more recently modified local copy (use --force to force download)


In [ ]:
! unzip nyc-property-sales.zip

Archive:  nyc-property-sales.zip
replace nyc-rolling-sales.csv? [y]es, [n]o, [A]ll, [N]one, [r]ename: y
  inflating: nyc-rolling-sales.csv   


# **1. Data Loading/Understanding**
Import Libraries yang dibutuhkan




In [ ]:
import pandas as pd
import numpy as np
import math
import statistics
import scipy.stats
import matplotlib.pyplot as plt

memanggil dan Menampilkan data yang ada didalam dataset

In [ ]:
df = pd.read_csv('nyc-rolling-sales.csv')
pd.set_option('display.max_columns', None)
df.head(5)

Unnamed: 0  BOROUGH   NEIGHBORHOOD  \
0           4        1  ALPHABET CITY   
1           5        1  ALPHABET CITY   
2           6        1  ALPHABET CITY   
3           7        1  ALPHABET CITY   
4           8        1  ALPHABET CITY   

                       BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  BLOCK  \
0  07 RENTALS - WALKUP APARTMENTS                                2A    392   
1  07 RENTALS - WALKUP APARTMENTS                                 2    399   
2  07 RENTALS - WALKUP APARTMENTS                                 2    399   
3  07 RENTALS - WALKUP APARTMENTS                                2B    402   
4  07 RENTALS - WALKUP APARTMENTS                                2A    404   

   LOT EASE-MENT BUILDING CLASS AT PRESENT                 ADDRESS  \
0    6                                  C2            153 AVENUE B   
1   26                                  C7   234 EAST 4TH   STREET   
2   39                                  C7   197 EAST 3RD   STREET   
3   21                                  C4     154 EAST 7TH STREET   
4   55                                  C2  301 EAST 10TH   STREET   

  APARTMENT NUMBER  ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  \
0                      10009                  5                 0   
1                      10009                 28                 3   
2                      10009                 16                 1   
3                      10009                 10                 0   
4                      10009                  6                 0   

   TOTAL UNITS LAND SQUARE FEET GROSS SQUARE FEET  YEAR BUILT  \
0            5             1633              6440        1900   
1           31             4616             18690        1900   
2           17             2212              7803        1900   
3           10             2272              6794        1913   
4            6             2369              4615        1900   

   TAX CLASS AT TIME OF SALE BUILDING CLASS AT TIME OF SALE SALE PRICE  \
0                          2                             C2    6625000   
1                          2                             C7        -     
2                          2                             C7        -     
3                          2                             C4    3936272   
4                          2                             C2    8000000   

             SALE DATE  
0  2017-07-19 00:00:00  
1  2016-12-14 00:00:00  
2  2016-12-09 00:00:00  
3  2016-09-23 00:00:00  
4  2016-11-17 00:00:00

In [ ]:
df.tail(5)

Unnamed: 0  BOROUGH NEIGHBORHOOD  \
84543        8409        5      WOODROW   
84544        8410        5      WOODROW   
84545        8411        5      WOODROW   
84546        8412        5      WOODROW   
84547        8413        5      WOODROW   

                           BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  \
84543  02 TWO FAMILY DWELLINGS                                        1   
84544  02 TWO FAMILY DWELLINGS                                        1   
84545  02 TWO FAMILY DWELLINGS                                        1   
84546  22 STORE BUILDINGS                                             4   
84547  35 INDOOR PUBLIC AND CULTURAL FACILITIES                       4   

       BLOCK  LOT EASE-MENT BUILDING CLASS AT PRESENT                ADDRESS  \
84543   7349   34                                  B9          37 QUAIL LANE   
84544   7349   78                                  B9       32 PHEASANT LANE   
84545   7351   60                                  B2       49 PITNEY AVENUE   
84546   7100   28                                  K6  2730 ARTHUR KILL ROAD   
84547   7105  679                                  P9      155 CLAY PIT ROAD   

      APARTMENT NUMBER  ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  \
84543                      10309                  2                 0   
84544                      10309                  2                 0   
84545                      10309                  2                 0   
84546                      10309                  0                 7   
84547                      10309                  0                 1   

       TOTAL UNITS LAND SQUARE FEET GROSS SQUARE FEET  YEAR BUILT  \
84543            2             2400              2575        1998   
84544            2             2498              2377        1998   
84545            2             4000              1496        1925   
84546            7           208033             64117        2001   
84547            1            10796              2400        2006   

       TAX CLASS AT TIME OF SALE BUILDING CLASS AT TIME OF SALE SALE PRICE  \
84543                          1                             B9     450000   
84544                          1                             B9     550000   
84545                          1                             B2     460000   
84546                          4                             K6   11693337   
84547                          4                             P9      69300   

                 SALE DATE  
84543  2016-11-28 00:00:00  
84544  2017-04-21 00:00:00  
84545  2017-07-05 00:00:00  
84546  2016-12-21 00:00:00  
84547  2016-10-27 00:00:00

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84548 entries, 0 to 84547
Data columns (total 22 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Unnamed: 0                      84548 non-null  int64 
 1   BOROUGH                         84548 non-null  int64 
 2   NEIGHBORHOOD                    84548 non-null  object
 3   BUILDING CLASS CATEGORY         84548 non-null  object
 4   TAX CLASS AT PRESENT            84548 non-null  object
 5   BLOCK                           84548 non-null  int64 
 6   LOT                             84548 non-null  int64 
 7   EASE-MENT                       84548 non-null  object
 8   BUILDING CLASS AT PRESENT       84548 non-null  object
 9   ADDRESS                         84548 non-null  object
 10  APARTMENT NUMBER                84548 non-null  object
 11  ZIP CODE                        84548 non-null  int64 
 12  RESIDENTIAL UNITS               84548 non-null

In [ ]:
# Konversi 'SALE PRICE' ke tipe data numerik
df['SALE PRICE']=df['SALE PRICE'].replace(' -  ', pd.NA)
df['SALE PRICE'] = pd.to_numeric(df['SALE PRICE'], errors='coerce')
df['SALE PRICE']=df['SALE PRICE'].fillna(0)
df['SALE PRICE'] = df['SALE PRICE'].astype('int64')

In [ ]:
df.describe()

Unnamed: 0       BOROUGH         BLOCK           LOT      ZIP CODE  \
count  84548.000000  84548.000000  84548.000000  84548.000000  84548.000000   
mean   10344.359878      2.998758   4237.218976    376.224015  10731.991614   
std     7151.779436      1.289790   3568.263407    658.136814   1290.879147   
min        4.000000      1.000000      1.000000      1.000000      0.000000   
25%     4231.000000      2.000000   1322.750000     22.000000  10305.000000   
50%     8942.000000      3.000000   3311.000000     50.000000  11209.000000   
75%    15987.250000      4.000000   6281.000000   1001.000000  11357.000000   
max    26739.000000      5.000000  16322.000000   9106.000000  11694.000000   

       RESIDENTIAL UNITS  COMMERCIAL UNITS   TOTAL UNITS    YEAR BUILT  \
count       84548.000000      84548.000000  84548.000000  84548.000000   
mean            2.025264          0.193559      2.249184   1789.322976   
std            16.721037          8.713183     18.972584    537.344993   
min             0.000000          0.000000      0.000000      0.000000   
25%             0.000000          0.000000      1.000000   1920.000000   
50%             1.000000          0.000000      1.000000   1940.000000   
75%             2.000000          0.000000      2.000000   1965.000000   
max          1844.000000       2261.000000   2261.000000   2017.000000   

       TAX CLASS AT TIME OF SALE    SALE PRICE  
count               84548.000000  8.454800e+04  
mean                    1.657485  1.056623e+06  
std                     0.819341  1.038794e+07  
min                     1.000000  0.000000e+00  
25%                     1.000000  0.000000e+00  
50%                     2.000000  4.150000e+05  
75%                     2.000000  8.300000e+05  
max                     4.000000  2.210000e+09

pengecekan kolom yang ada di dataset untuk mengetahui apakah ada nilai null didalam kolom.

In [ ]:
# Count the number of 0 values in the 'SALE PRICE' column
sum = (df['SALE PRICE'] == 0).sum()
sum

59759

In [ ]:
df.isnull().sum()

Unnamed: 0                        0
BOROUGH                           0
NEIGHBORHOOD                      0
BUILDING CLASS CATEGORY           0
TAX CLASS AT PRESENT              0
BLOCK                             0
LOT                               0
EASE-MENT                         0
BUILDING CLASS AT PRESENT         0
ADDRESS                           0
APARTMENT NUMBER                  0
ZIP CODE                          0
RESIDENTIAL UNITS                 0
COMMERCIAL UNITS                  0
TOTAL UNITS                       0
LAND SQUARE FEET                  0
GROSS SQUARE FEET                 0
YEAR BUILT                        0
TAX CLASS AT TIME OF SALE         0
BUILDING CLASS AT TIME OF SALE    0
SALE PRICE                        0
SALE DATE                         0
dtype: int64

dilihat pada kode diatas, pada kolom/feature "SALE PRICE" terdapat nilai null sebanyak 14561. ini merupakan nilai yang cukup besar sehingga dapat memengaruhi hasil analisis kita nanti sehingga ini perlu di treatment.

In [ ]:
mean_price = df['SALE PRICE'].mean()
df['SALE PRICE'] = df['SALE PRICE'].replace(0, mean_price)

In [ ]:
df

Unnamed: 0  BOROUGH   NEIGHBORHOOD  \
0               4        1  ALPHABET CITY   
1               5        1  ALPHABET CITY   
2               6        1  ALPHABET CITY   
3               7        1  ALPHABET CITY   
4               8        1  ALPHABET CITY   
...           ...      ...            ...   
84543        8409        5        WOODROW   
84544        8410        5        WOODROW   
84545        8411        5        WOODROW   
84546        8412        5        WOODROW   
84547        8413        5        WOODROW   

                           BUILDING CLASS CATEGORY TAX CLASS AT PRESENT  \
0      07 RENTALS - WALKUP APARTMENTS                                2A   
1      07 RENTALS - WALKUP APARTMENTS                                 2   
2      07 RENTALS - WALKUP APARTMENTS                                 2   
3      07 RENTALS - WALKUP APARTMENTS                                2B   
4      07 RENTALS - WALKUP APARTMENTS                                2A   
...                                            ...                  ...   
84543  02 TWO FAMILY DWELLINGS                                        1   
84544  02 TWO FAMILY DWELLINGS                                        1   
84545  02 TWO FAMILY DWELLINGS                                        1   
84546  22 STORE BUILDINGS                                             4   
84547  35 INDOOR PUBLIC AND CULTURAL FACILITIES                       4   

       BLOCK  LOT EASE-MENT BUILDING CLASS AT PRESENT                 ADDRESS  \
0        392    6                                  C2            153 AVENUE B   
1        399   26                                  C7   234 EAST 4TH   STREET   
2        399   39                                  C7   197 EAST 3RD   STREET   
3        402   21                                  C4     154 EAST 7TH STREET   
4        404   55                                  C2  301 EAST 10TH   STREET   
...      ...  ...       ...                       ...                     ...   
84543   7349   34                                  B9           37 QUAIL LANE   
84544   7349   78                                  B9        32 PHEASANT LANE   
84545   7351   60                                  B2        49 PITNEY AVENUE   
84546   7100   28                                  K6   2730 ARTHUR KILL ROAD   
84547   7105  679                                  P9       155 CLAY PIT ROAD   

      APARTMENT NUMBER  ZIP CODE  RESIDENTIAL UNITS  COMMERCIAL UNITS  \
0                          10009                  5                 0   
1                          10009                 28                 3   
2                          10009                 16                 1   
3                          10009                 10                 0   
4                          10009                  6                 0   
...                ...       ...                ...               ...   
84543                      10309                  2                 0   
84544                      10309                  2                 0   
84545                      10309                  2                 0   
84546                      10309                  0                 7   
84547                      10309                  0                 1   

       TOTAL UNITS LAND SQUARE FEET GROSS SQUARE FEET  YEAR BUILT  \
0                5             1633              6440        1900   
1               31             4616             18690        1900   
2               17             2212              7803        1900   
3               10             2272              6794        1913   
4                6             2369              4615        1900   
...            ...              ...               ...         ...   
84543            2             2400              2575        1998   
84544            2             2498              2377        1998   
84545            2             4000              1496        1925   
84546            7       

In [ ]:
df.columns

Index(['Unnamed: 0', 'BOROUGH', 'NEIGHBORHOOD', 'BUILDING CLASS CATEGORY',
       'TAX CLASS AT PRESENT', 'BLOCK', 'LOT', 'EASE-MENT',
       'BUILDING CLASS AT PRESENT', 'ADDRESS', 'APARTMENT NUMBER', 'ZIP CODE',
       'RESIDENTIAL UNITS', 'COMMERCIAL UNITS', 'TOTAL UNITS',
       'LAND SQUARE FEET', 'GROSS SQUARE FEET', 'YEAR BUILT',
       'TAX CLASS AT TIME OF SALE', 'BUILDING CLASS AT TIME OF SALE',
       'SALE PRICE', 'SALE DATE'],
      dtype='object')

In [ ]:
# Membuat mapping antara nilai baru dan nilai asli borough
mapping_borough = {
    1: 'Manhattan',
    2: 'Bronx',
    3: 'Brooklyn',
    4: 'Queens',
    5: 'Staten Island'
}

# Mengganti nilai kolom 'BOROUGH' dengan nilai asli menggunakan mapping
df['BOROUGH'] = df['BOROUGH'].map(mapping_borough)

Seletah melalui Data Understanding, kami ingin menemukan insight yang ada didalam data ini. kami ingin menggunakan measure of central tendency seperti mean, median,modus,range, variance,standard Deviation,Probability distribution. kami ingin menggunakan mean untuk mengetahui rata rata tahun pembangunan gedung di masing-masing borough.

In [ ]:
rata_rata_per_borough = df.groupby('BOROUGH')['YEAR BUILT'].mean()
print("Rata-rata tahun pembangunan properti adalah:", rata_rata_per_borough)

Rata-rata tahun pembangunan properti adalah: BOROUGH
Bronx            1745.715279
Brooklyn         1767.358673
Manhattan        1706.773845
Queens           1852.045631
Staten Island    1868.961237
Name: YEAR BUILT, dtype: float64


Berdasarkan nilai rata rata diatas, dapat diketahui bahwa pada daerah Bronx, rata rata gedung yang dibangun pada tahun 1745, pada Brooklyn tahun 1767, dll.

kami juga ingin mengetahui pada tanggal bulan dan tahun berapa terjadi transaksi terbanyak dengan memanfaatkan modus dalam mencarinya.

In [ ]:
# Menghitung modus dari kolom 'SALE DATE'
modus_sale_date = df['SALE DATE'].mode()

# Menghitung frekuensi kemunculan setiap tanggal
frekuensi_tanggal = df['SALE DATE'].dt.date.value_counts()

# Menemukan tanggal yang menjadi modus
tanggal_modus = frekuensi_tanggal.idxmax()

# Menampilkan tanggal yang menjadi modus dan frekuensi kemunculannya
print("Tanggal Modus SALE DATE:", tanggal_modus)
print("Frekuensi Kemunculan:", frekuensi_tanggal.max())

Tanggal Modus SALE DATE: 2017-06-29
Frekuensi Kemunculan: 544


Kemudian, saya ingin mengetahui berapa jumlah transaksi terbesar terjadi pada bulan dan tanggal berapa selama dari awal 2016 hingga akhir tahun 2016

In [ ]:
# Tentukan range (misalnya, Januari 2023 sampai Maret 2023)
start_date = pd.to_datetime('2016-01-01')
end_date = pd.to_datetime('2016-12-31')

# Mengubah kolom 'SALE DATE' ke tipe data datetime
df['SALE DATE'] = pd.to_datetime(df['SALE DATE'], errors='coerce')
# Filter dataset berdasarkan range tanggal
filtered_df = df[(df['SALE DATE'] >= start_date) & (df['SALE DATE'] <= end_date)]


# Periksa apakah ada data yang tersedia dalam rentang waktu
if not filtered_df.empty:
    # Menghitung frekuensi kemunculan setiap tanggal dalam range
    frekuensi_tanggal_range = filtered_df['SALE DATE'].dt.date.value_counts()

    # Menemukan tanggal yang menjadi modus dalam range
    tanggal_modus_range = frekuensi_tanggal_range.idxmax()

    # Menghitung total transaksi pada tanggal modus
    total_transaksi_modus = filtered_df[filtered_df['SALE DATE'].dt.date == tanggal_modus_range]['SALE PRICE'].count()

    # Menampilkan tanggal yang menjadi modus, frekuensi kemunculan, dan total transaksi pada tanggal modus
    print("Tanggal Modus SALE DATE (dalam range):", tanggal_modus_range)
    print("Frekuensi Kemunculan (dalam range):", frekuensi_tanggal_range.max())
    print("Total Transaksi pada Tanggal Modus:", total_transaksi_modus)
else:
    print("Tidak ada data yang tersedia dalam rentang waktu yang diberikan.")


Tanggal Modus SALE DATE (dalam range): 2016-12-22
Frekuensi Kemunculan (dalam range): 527
Total Transaksi pada Tanggal Modus: 527


diketahui dari data diatas, tingkat transaksi terbanyak terjadi pada 22 Desember 2016 dengan total transaksi terjadi sebanyak 527 kali.

In [ ]:
# Konversi 'SALE PRICE' ke tipe data numerik (jika mungkin)
df['SALE PRICE'] = pd.to_numeric(df['SALE PRICE'], errors='coerce')
sales_price = df['SALE PRICE']
# 4. Range
range_sales_price = np.max(sales_price) - np.min(sales_price)
print(f"Range Harga Penjualan pada sales price adalah: {range_sales_price}")

Range Harga Penjualan pada sales price adalah: 2210000000.0


Range Harga Penjualan: Nilai ini menunjukkan selisih antara harga penjualan tertinggi dan terendah. Semakin besar nilai range, semakin besar variasi harga penjualan.

In [ ]:
sales_price

0         6625000.0
1               NaN
2               NaN
3         3936272.0
4         8000000.0
            ...    
84543      450000.0
84544      550000.0
84545      460000.0
84546    11693337.0
84547       69300.0
Name: SALE PRICE, Length: 84548, dtype: float64

In [ ]:
# 5. Variance
variance_sales_price = np.var(sales_price)
print(f"Varians Harga Penjualan pada sales price adalah: {variance_sales_price:.2f}")

Varians Harga Penjualan pada sales price adalah: 130077990878434.38


Varians Harga Penjualan: Varians mengukur seberapa jauh setiap nilai harga penjualan dari rata-rata. Varians yang tinggi menunjukkan bahwa harga penjualan tersebar lebih jauh dari rata-rata.

In [ ]:
# 6. Standard Deviation
std_dev_sales_price = np.std(sales_price)
print(f"Deviasi Standar Harga Penjualan pada sales price adalah: {std_dev_sales_price:.2f}")

Deviasi Standar Harga Penjualan pada sales price adalah: 11405173.86


Deviasi Standar Harga Penjualan: Deviasi standar adalah akar kuadrat dari varians. Nilai ini memberikan gambaran yang lebih mudah diinterpretasikan tentang sebaran harga penjualan. Semakin tinggi deviasi standar, semakin besar variasi harga penjualan.

## **Conclusion**

Berdasarkan data diatas, kita dapat mengetahui bahwa:
1. Rata-rata gedung dibangun berdasarkan dataset diatas per borough:
    * Bronx rata-rata padah tahun 1745
    * Brooklyn rata-rata padah tahun 1767
    * Manhattan rata-rata padah tahun 1706
    * Queens rata-rata padah tahun 1852
    * Staten Island rata-rata padah tahun 1868

2. Tanggal dimana penjualan terbanyak terjadi pada 2017-06-29 dengan frekuensi  Kemunculan sebanyak 544 kali

3. Range Harga Penjualan pada sales price adalah: 2210000000 dimana Range Harga Penjualan: Nilai ini menunjukkan selisih antara harga penjualan tertinggi dan terendah. Semakin besar nilai range, semakin besar variasi harga penjualan.

4. Varians Harga Penjualan pada sales price adalah: 130077990878434.38
Varians mengukur seberapa jauh setiap nilai harga penjualan dari rata-rata. Varians yang tinggi menunjukkan bahwa harga penjualan tersebar lebih jauh dari rata-rata.
5. eviasi Standar Harga Penjualan pada sales price adalah: 11405173.86
 Deviasi standar adalah akar kuadrat dari varians. Nilai ini memberikan gambaran yang lebih mudah diinterpretasikan tentang sebaran harga penjualan. Semakin tinggi deviasi standar, semakin besar variasi harga penjualan.